File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/SwitchSectionShouldNotHaveTooManyStatements_DefaultValue.cs
using System;

namespace Tests.Diagnostics
{
    public class Program
    {
        public Program()
        {
            switch (myVariable)
            {
                case 0:
                    break;
                case 1: // Noncompliant {{Reduce this 'switch/case' number of lines from 6 to at most 5, for example by extracting code into a method.}}
//              ^^^^^^^
                    Console.WriteLine("1");
                    Console.WriteLine("2");
                    Console.WriteLine("3");
                    Console.WriteLine("4");
                    Console.WriteLine("5");
                    break;
                case 2:
                    {
                        Console.WriteLine("1");
                        Console.WriteLine("2");
                        Console.WriteLine("3");
                        Console.WriteLine("4");
                        break;
                    }
                case 3: // Noncompliant {{Reduce this 'switch/case' number of lines from 6 to at most 5, for example by extracting code into a method.}}
                case 4:
                    if (true)
                    {
                        if (false)
                        {
                            for (int i = 0; i < length; i++)
                            {
                                Console.WriteLine();
                                Console.WriteLine();
                                Console.WriteLine();
                            }
                        }
                    }
                case 5: // Noncompliant
                    var x = Foo(42);
                    break;

                    string Foo(int value)
                    {
                        Console.WriteLine("1");
                        Console.WriteLine("2");
                        Console.WriteLine("3");
                        Console.WriteLine("4");
                        Console.WriteLine("5");
                    }
                case:
                default:
                    break;
            }
        }
    }
}

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/SwitchSectionShouldNotHaveTooManyStatements_DefaultValue.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Microsoft.VisualBasic

Namespace Tests.TestCases
    Class Foo
        Public Sub Test()
            Dim number As Integer = 8

            Select Case number
                Case 1 To 5 ' Noncompliant {{Reduce this 'Case' block number of statements from 9 to at most 8, for example by extracting code into a procedure.}}
'               ^^^^^^^^^^^
                    Test()
                    Test()
                    Test()
                    Test()
                    Test()
                    Test()
                    Test()
                    Test()
                    Test()
                Case 6 ' Noncompliant {{Reduce this 'Case' block number of statements from 12 to at most 8, for example by extracting code into a procedure.}}
                    If False
                    ElseIf True
                    Else
                        For index = 1 To 10
                        Next
                        While True
                        End While
                        Do
                        Loop While (number < 1)
                        Do
                        Loop Until (number = 1)
                        Select Case number
                            Case 1
                            Case 2
                            Case Else
                        End Select
                        If True Then Test()
                    End If
                Case Else
                    Test()
            End Select
		End Sub
    End Class
End Namespace
